### PR TITLE
[react-calendar-heatmap] Reduce Usage of 'any' Type

### PR DESCRIPTION
**Pull request by penpendayo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-calendar-heatmap/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bfbbdb98a127f01dc5e6fa7c46d306248dca1e07
</p>

File: types/react-calendar-heatmap/index.d.ts [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
     showOutOfRangeDays?: boolean | undefined;
     showWeekdayLabels?: boolean | undefined;
     startDate?: string | number | Date | undefined;
-    titleForValue?: ((value: any) => any) | undefined;
+    titleForValue?: ((value: ReactCalendarHeatmapValue<T> | undefined) => string) | undefined;
     tooltipDataAttrs?: object | undefined;
-    transformDayElement?: ((rect: any, value: any, index: number) => any) | undefined;
-    values: any[];
+    transformDayElement?: ((element: React.ReactElement, value: ReactCalendarHeatmapValue<T> | undefined, index: number) => React.ReactNode) | undefined;
+    values: Array<ReactCalendarHeatmapValue<T>>;
```
Comment (9renpoto) [inline types/react-calendar-heatmap/index.d.ts:26]: https://github.com/kevinsqi/react-calendar-heatmap/blob/v1.8.1/src/index.js#L342C1-L347

OK

File: types/react-calendar-heatmap/index.d.ts [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
-    onMouseOver?: ((e: any, value: any) => void) | undefined;
+    onClick?: ((value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
+    onMouseLeave?: ((e: React.MouseEvent<SVGRectElement, MouseEvent>, value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
+    onMouseOver?: ((e: React.MouseEvent<SVGRectElement, MouseEvent>, value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
     showMonthLabels?: boolean | undefined;
     showOutOfRangeDays?: boolean | undefined;
     showWeekdayLabels?: boolean | undefined;
     startDate?: string | number | Date | undefined;
-    titleForValue?: ((value: any) => any) | undefined;
+    titleForValue?: ((value: ReactCalendarHeatmapValue<T> | undefined) => string) | undefined;
```
Comment (9renpoto) [inline types/react-calendar-heatmap/index.d.ts:23]: https://github.com/kevinsqi/react-calendar-heatmap/blob/v1.8.1/src/index.js#L101-L108

OK

File: types/react-calendar-heatmap/index.d.ts [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
     gutterSize?: number | undefined;
     horizontal?: boolean | undefined;
     monthLabels?: string[] | undefined;
     numDays?: number | undefined;
-    onClick?: ((value: any) => void) | undefined;
-    onMouseLeave?: ((e: any, value: any) => void) | undefined;
-    onMouseOver?: ((e: any, value: any) => void) | undefined;
+    onClick?: ((value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
+    onMouseLeave?: ((e: React.MouseEvent<SVGRectElement, MouseEvent>, value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
+    onMouseOver?: ((e: React.MouseEvent<SVGRectElement, MouseEvent>, value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
```
Comment (9renpoto) [inline types/react-calendar-heatmap/index.d.ts:18]: https://github.com/kevinsqi/react-calendar-heatmap/blob/v1.8.1/src/index.js#L239-L250

OK

File: types/react-calendar-heatmap/index.d.ts [hunk 4]
```diff
@@ -1,24 +1,30 @@
...
     endDate?: string | number | Date | undefined;
     gutterSize?: number | undefined;
     horizontal?: boolean | undefined;
     monthLabels?: string[] | undefined;
     numDays?: number | undefined;
-    onClick?: ((value: any) => void) | undefined;
-    onMouseLeave?: ((e: any, value: any) => void) | undefined;
-    onMouseOver?: ((e: any, value: any) => void) | undefined;
+    onClick?: ((value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
+    onMouseLeave?: ((e: React.MouseEvent<SVGRectElement, MouseEvent>, value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
```
Comment (9renpoto) [inline types/react-calendar-heatmap/index.d.ts:17]: https://github.com/kevinsqi/react-calendar-heatmap/blob/v1.8.1/src/index.js#L239-L251

OK

File: types/react-calendar-heatmap/index.d.ts [hunk 5]
```diff
@@ -1,24 +1,30 @@
...
+    classForValue?: ((value: ReactCalendarHeatmapValue<T> | undefined) => string) | undefined;
     endDate?: string | number | Date | undefined;
     gutterSize?: number | undefined;
     horizontal?: boolean | undefined;
     monthLabels?: string[] | undefined;
     numDays?: number | undefined;
-    onClick?: ((value: any) => void) | undefined;
-    onMouseLeave?: ((e: any, value: any) => void) | undefined;
-    onMouseOver?: ((e: any, value: any) => void) | undefined;
+    onClick?: ((value: ReactCalendarHeatmapValue<T> | undefined) => void) | undefined;
```
Comment (9renpoto) [inline types/react-calendar-heatmap/index.d.ts:16]: https://github.com/kevinsqi/react-calendar-heatmap/blob/v1.8.1/src/index.js#L239-L249C34

OK

**Review by 9renpoto — APPROVED** [review]
LGTM

**Comment by 9renpoto** [comment]
Ready to merge

**Pull request merged**